m4/nut_check_libopenssl.m4

```diff
@@ -58,7 +58,7 @@ if test -z "${nut_have_libopenssl_seen}"; then
 
 	dnl check if openssl is usable
 	AC_CHECK_HEADERS(openssl/ssl.h, [nut_have_openssl=yes], [nut_have_openssl=no], [AC_INCLUDES_DEFAULT])
-	AC_CHECK_FUNCS(SSL_library_init, [], [nut_have_openssl=no])
+	AC_CHECK_FUNCS(SSL_CTX_new, [], [nut_have_openssl=no])
```

Comment:
Just in case : Is this symbol available in all openssl versions?

If not, maybe its presence vs. the older symbol can be used to differentiate behaviors somehow (as a measure of codebase age)...

Reply:
SSL_CTX_new is available since at least 0.9.6.  My goal was to find a common-ground function.

Reply:
It's been available since SSLeay 0.5.1 (the predecessor of OpenSSL), so it's available everywhere.

One should only look at OPENSSL_VERSION_NUMBER to determine if some function is available, the check for SSL_CTX_new is only there to make sure if the libraries are installed properly.

Reply:
Using the TLSv1_client_method was a quick-and-dirty way to disable SSLv2 and SSLv3 and really never the way to do this. Look at PR#504 for the proper way. You really shouldn't use TLSv1 if both server and client support TLSv1.2.

Reply:
Also, PR #504 also documents why we did this back then. The code has been slowly rotting in the tree until OpenSSL 1.1.0 came out, by which time I was no longer involved in active development.

Reply:
I never meat this to replace #504, since it is more complete than this one.  I just saw it open for a while, and tried a slimmer version in the hope to get it to at least compile with openssl 1.1.x.  By all means, your idea is right on.

Reply:
I'll close this, and add some comments on #504.  This is the only one change I think could be used the way I wrote, as it does its job and is simpler/faster.

Reply:
I agree, checking for SSL_CTX_new is probably less intrusive and should be enough to make sure that not only the headers are available, but one can actually link to OpenSSL as well. I remember this was a frequent source of problems.